Mathlib/Topology/CompletelyRegular.lean

```diff
@@ -0,0 +1,109 @@
+/-
+Copyright (c) 2032 Matias Heikkilä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Matias Heikkilä
+-/
+import Mathlib.Topology.UrysohnsLemma
+import Mathlib.Topology.UnitInterval
+
+/-!
+# Completely regular topological spaces.
+
+This file defines `CompletelyRegularSpace`.
+
+## Main definitions
+
+* `CompletelyRegularSpace`: A completely regular space `X` is a T₁ space where, for each
+  closed set `K ⊆ X` and a point `x ∈ Kᶜ`, there is a continuous function `f` from
+  `X` to the unit interval, so that `f x = 0` and `f k = 1` for all `k ∈ K`. A completely
+  regular space is a T₂ space and a T₄ space is completely regular.
+
+## Main results
+
+### Completely regular spaces
+
+* `CompletelyRegularSpace`: Given a point `x` and a closed set `K` such that `{x} ∩ K = ∅ `,
+  we can find a continuous real-valued function `f` such that `f` is equal to `0` on `{x}`
+  and `f` is equal to `1` on `K`.
+* `CompletelyRegularSpace.t3space`: A completely regular space is a T₃ space.
+* `NormalSpace.completelyRegularSpace`: A normal space is a completely regular space.
+
+## Implementation notes
+
+The present definition `CompletelyRegularSpace` is a slight modification of the one given in
+[russell1974]. There it's assumed that any point `x ∈ Kᶜ` is separated from the closed set `K` by a
+continuous *real* valued function `f` (as opposed to `f` being unit-interval-valued). This can be
+converted to the present definition by replacing a real-valued `f` by `h ∘ g ∘ f`, with
+`g : x ↦ max(x, 0)` and `h : x ↦ min(x, 1)`.
+
+## References
+
+* [Russell C. Walker, *The Stone-Čech Compactification*][russell1974]
+-/
+
+universe u
+
+noncomputable section
+
+open Filter Set Topology unitInterval Set.Icc
+
+variable {X : Type u} [TopologicalSpace X] [T1Space X]
+
+/-- A T1 space X is completely regular if points can be separated from closed sets via
+  continuous functions to the unit interval. -/
+@[mk_iff completelyRegularSpace_iff]
+class CompletelyRegularSpace (X : Type u) [TopologicalSpace X] [T1Space X] : Prop where
+  completely_regular : ∀ (x : X), ∀ (K : Set X) (_: IsClosed K), x ∉ K →
+    ∃ f : X → I, Continuous f ∧ f x = 0 ∧ EqOn f 1 K
+
+instance CompletelyRegularSpace.t3space [CompletelyRegularSpace X] : T3Space X := by
+  have : T0Space X := by apply T1Space.t0Space
+  have : RegularSpace X := by
+    have ne : (0 : I) ≠ 1 := by
```

Comment:
Is there a one-liner for this somewhere?

Reply:
[one_ne_zero](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/NeZero.html#one_ne_zero).symm should work